### PR TITLE
Implement Edit, Delete Views

### PR DESCRIPTION
**Pull request by lawsonbh**
Starting to convert some of the views (Ego) over to classes to implement edit and delete functionality

### CODE REVIEW

**Comment by lawsonbh** [comment]
Ok, questions for @bbelderbos :

- How can I tell where I will bring in a template from? I am getting thrown off in reasoning in my code. The django error messages help but I think I am missing something in views.py of character_generator - I wanted to specify each template path in the template_name vars as being "character_generator/form.html" but instead found that just "form.html" worked. Why? :raccoon: 
- Mixins were really confusing at first but working through the ego edit and list views really made me understand the intuitiveness of it on the other side of implementing it. That being said . . . would be doing this for multiple models? Or I guess - would we do this for morph and items as well? I could see these views.py files getting really large. How does this work on a project with many models?
- How do I make sure the user needs to be logged in with a class based view? Can I just add the decorator @login_required above it? Or is there something else I should do? Getting an anon user error when not logged which is ok - but I'd like to keep it cleaner.
- Can we work together on removing the function based views we are no longer using? Wanted to run this all by you before removing them.

**Comment by lawsonbh** [comment]
If we like this approach, this PR introduces Ego list view, Ego edit view, Ego create view (version 2 :smile: ), and Ego delete view and will close #7

File: bodybank/character_generator/views.py [hunk 1]
```diff
@@ -23,3 +27,35 @@ def get_ego(request):
 
 def say_thanks(request):
     return render(request, 'thanks.html')
+
+class OwnerMixin():
```
Comment (bbelderbos) [inline bodybank/character_generator/views.py]: It's a good exercise for Django devs to at least once use class based views. They save code, but if you need to do something a bit more complicated it becomes confusing soon!

Contrast this with the relative simplicity of function based views.

I buy into much of what Luke Plant says here:
https://spookylukey.github.io/django-views-the-right-way/

I wrapped the most important points into a preso:
https://docs.google.com/presentation/d/1GBp6xJtxcQi2r2MHc1Mn4a8Bkxkuzt9pmckY1LAzhio/edit#slide=id.g11fa00ee09f_0_10





[7:22](https://pybitesdevelo-ddf2687.slack.com/archives/D03S0PS3067/p1663824179665649)

If you prefer video you can watch the Code Clinic as well: https://pybites.mykajabi.com/products/pybites-coaching-calls-2/categories/3469727/posts/2156867625

File: bodybank/character_generator/templates/list.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{% extends "base.html" %}
+
+{% block title %}My egos{% endblock %}
+
+{% block content %}
+  <h1>My egos</h1>
+  <div class="module">
+    {% for ego in object_list %}
```
Comment (bbelderbos) [inline bodybank/character_generator/templates/list.html]: See comment below, for example to me it's not obvious that class based views pass on an object_list to the template, it's not transparent. With a function based view you're very explicit with what you pass through to the template via the context dictionary you pass into the render() function. So to me function based views are more explicit than class based views (Zen of Python!). Then there is the multiple levels of inheritance and hence the need for resources like this: https://ccbv.co.uk
I am not using them anymore :)
But (!) it's really good you took knowledge of them!

**Review by bbelderbos — COMMENTED** [review]
My opinionated review :)

**Comment by bbelderbos** [comment]
> 

I think most questions are typical / common when using class based views. If you don't mind let's have our future selves thank us and let's switch to function based views.

Of course don't take my word for it yet. Please look at the clinic presentation and let me know if it makes sense after that ...

HTH
Bob

File: bodybank/character_generator/templates/list.html (hunk 2 above)
Comment (lawsonbh) [reply bodybank/character_generator/templates/list.html]: I really like this excerpt: The fundamental nature of a view is that it is a function (or callable) that takes a request and returns a response.

We can switch over to function based views. They were more straightforward to work with.

**Comment by lawsonbh** [comment]
Ok! So going back to function based views b/c lord there are a lot of classes to track! :raccoon: 

So instead of this
`Can we work together on removing the function based views we are no longer using? Wanted to run this all by you before removing them.`

We will remove the class based views and replace them with function based views so we still have a view for:
1. List of User Egos (limit to active user)
2. edit ego
3. delete ego

**Pull request merged**